use full paths to irssi includes

ailin-nemui opened the pull request:
this is in parallel to the change cdidier/irssi-xmpp#51 and irssi/irssi#1040 for irssi 1.3

singpolyma commented:
Seems reasonable.  I should probably get back to this fork in general...

janLo commented:
I recommend to use GpgME for the pgp stuff. Also the use of tmpnam is not the best option:

> Warning: Between the time the pathname is constructed and the file is created another process 
> might have created a file with the same name using tmpnam, leading to a possible security hole. The 
> implementation generates names which can hardly be predicted, but when opening the file you 
> should use the O_EXCL flag. Using tmpfile or mkstemp is a safe way to avoid this problem.

Another problem is, that my /tmp was filled up because the tmpfiles where not cleaned up. At least you can try to switch to use gcc extensions for ressource handling:
http://en.wikipedia.org/wiki/Resource_Acquisition_Is_Initialization#GCC_.22cleanup.22_extension_for_C

singpolyma commented:
This whole thing was pretty hacked together. gpgme is probably the right way to go, since it probably handles TOCTOU, etc, mitigation better

singpolyma commented:
@janLo I have moved the old branch to old-master and made a new master based on the actual upstream git (since upstream has moved to git since this project started).  I have also done some code cleanup, including no longer using tempfiles for anything.

ailin-nemui committed:
use full paths to irssi includes

this is in parallel to the change cdidier/irssi-xmpp#51 and rssi/irssi#1040 for irssi 1.3
